### PR TITLE
LOOP-4781 Update LoopKit to use LoopAlgorithm swift package

### PR DESCRIPTION
**Pull request by ps2**

### CODE REVIEW

File: LoopKit/Extensions/Date.swift [hunk 1]
```diff
@@ -9,20 +9,20 @@
...
-
-    func dateCeiledToTimeInterval(_ interval: TimeInterval) -> Date {
-        if interval == 0 {
-            return self
-        }
-
-        return Date(timeIntervalSinceReferenceDate: ceil(self.timeIntervalSinceReferenceDate / interval) * interval)
-    }
-}
+//public extension Date {
```
Comment (Camji55) [inline LoopKit/Extensions/Date.swift]: Can this be deleted?

File: LoopKit/Extensions/HKQuantity.swift [hunk 2]
```diff
@@ -10,9 +10,9 @@ import Foundation
 import HealthKit
 
 
-extension HKQuantity: Comparable { }
-
-
-public func <(lhs: HKQuantity, rhs: HKQuantity) -> Bool {
-    return lhs.compare(rhs) == .orderedAscending
-}
+//extension HKQuantity: Comparable { }
```
Comment (Camji55) [inline LoopKit/Extensions/HKQuantity.swift]: Can this be deleted?

File: LoopKit/GlucoseChange.swift [hunk 3]
```diff
@@ -8,20 +8,20 @@
...
-    mutating public func append(_ effect: GlucoseEffect) {
-        startDate = min(effect.startDate, startDate)
-        endDate = max(effect.endDate, endDate)
-        quantity = HKQuantity(
-            unit: .milligramsPerDeciliter,
-            doubleValue: quantity.doubleValue(for: .milligramsPerDeciliter) + effect.quantity.doubleValue(for: .milligramsPerDeciliter)
-        )
-    }
-}
+//public struct GlucoseChange: SampleValue, Equatable {
```
Comment (Camji55) [inline LoopKit/GlucoseChange.swift]: Can this be deleted?

File: LoopKit/InsulinKit/AutomaticDoseRecommendation.swift [hunk 4]
```diff
@@ -8,18 +8,18 @@
...
-        self.bolusUnits = bolusUnits
-    }
-
-    public var hasDosingChange: Bool {
-        return basalAdjustment != nil || bolusUnits != nil
-    }
-}
-
-extension AutomaticDoseRecommendation: Codable {}
+//public struct AutomaticDoseRecommendation: Equatable {
```
Comment (Camji55) [inline LoopKit/InsulinKit/AutomaticDoseRecommendation.swift]: Can this be deleted?

File: LoopKit/InsulinKit/InsulinValue.swift [hunk 5]
```diff
@@ -8,33 +8,34 @@
...
-                return nil
-            }
-            if let end, entry.startDate > end {
-                return nil
-            }
-            return entry
-        }
-    }
-}
+//public struct InsulinValue: TimelineValue, Equatable {
```
Comment (Camji55) [inline LoopKit/InsulinKit/InsulinValue.swift]: Can this be deleted?

File: LoopKit/InsulinKit/ManualBolusRecommendation.swift [hunk 6]
```diff
@@ -8,107 +8,100 @@
...
-extension ManualBolusRecommendation: Comparable {
-    public static func ==(lhs: ManualBolusRecommendation, rhs: ManualBolusRecommendation) -> Bool {
-        return lhs.amount == rhs.amount
-    }
-
-    public static func <(lhs: ManualBolusRecommendation, rhs: ManualBolusRecommendation) -> Bool {
-        return lhs.amount < rhs.amount
-    }
-}
+//extension BolusRecommendationNotice: Codable {
```
Comment (Camji55) [inline LoopKit/InsulinKit/ManualBolusRecommendation.swift]: Can this be deleted?

File: LoopKit/InsulinKit/TempBasalRecommendation.swift [hunk 7]
```diff
@@ -9,23 +9,23 @@
...
-    }
-
-    public init(unitsPerHour: Double, duration: TimeInterval) {
-        self.unitsPerHour = unitsPerHour
-        self.duration = duration
-    }
-}
-
-extension TempBasalRecommendation: Codable {}
+//public struct TempBasalRecommendation: Equatable {
```
Comment (Camji55) [inline LoopKit/InsulinKit/TempBasalRecommendation.swift]: Can this be deleted?

File: LoopKitTests/Fixtures/InsulinKit/suspend_dose_reconciled_normalized.json [hunk 8]
```diff
@@ -3,7 +3,7 @@
         "type": "TempBasal",
         "start_at": "2015-07-13T12:00:00",
         "end_at": "2015-07-13T12:00:00",
-        "amount": -0.75,
+        "amount": 0,
```
Comment (Camji55) [inline LoopKitTests/Fixtures/InsulinKit/suspend_dose_reconciled_normalized.json:6]: Curious why this change is needed?

File: LoopKitTests/BolusRecommendationTests.swift [hunk 9]
```diff
@@ -128,20 +125,20 @@ class BolusRecommendationNoticeCodableTests: XCTestCase {
...
+//            return lhsGlucoseValue.startDate == rhsGlucoseValue.startDate &&
+//                lhsGlucoseValue.endDate == rhsGlucoseValue.endDate &&
+//                lhsGlucoseValue.quantity == rhsGlucoseValue.quantity
+//        case (.predictedGlucoseInRange, .predictedGlucoseInRange):
+//            return true
+//        default:
+//            return false
+//        }
+//    }
+//}
```
Comment (nhamming) [inline LoopKitTests/BolusRecommendationTests.swift:144]: remove?

**Review by nhamming — APPROVED** [review]

File: LoopKitTests/Fixtures/InsulinKit/suspend_dose_reconciled_normalized.json (hunk 8 above)
Comment (ps2) [reply LoopKitTests/Fixtures/InsulinKit/suspend_dose_reconciled_normalized.json:6]: It's related to how "reconciled and normalized" dose volumes are tracked in the new system: BasalRelativeDose now instead of a DoseEntry that has a scheduled basal rate added to it.  BasalRelativeDose uses *volume* to track total dose amount, not switching between *volume* and *rate* like DoseEntry does. The dose in question has a duration of 0, which means volume is 0.  This is the way suspend events are tracked. At some point we get a resume, and then that becomes a real dose.

This test is maybe ineffectual; it's a little confusing as to what it's even supposed to test; the resulting IOB array at the end of the test is all 0's because the suspend is 0 length; i.e. no insulin effect at all. Looking back in history, it was this way even back to Loop main, before any algo work. 

I think the test is still operating as it does on main, even with this change to `amount`. It would probably make more sense to update the test to assume the suspend had been running for some time, and then have some negative IOB in the final array.

**Review by Camji55 — APPROVED** [review]
